src/lib/sync_handler/sync_handler.ml

```diff
@@ -57,35 +67,38 @@ module Make (Inputs : Inputs_intf) :
   let get_staged_ledger_aux_and_pending_coinbases_at_hash ~frontier state_hash
       =
     let open Option.Let_syntax in
-    let%map breadcrumb =
+    let%map scan_state, pending_coinbases =
       Option.merge
-        (Transition_frontier.find frontier state_hash)
-        (Transition_frontier.find_in_root_history frontier state_hash)
+        (let%map breadcrumb = Transition_frontier.find frontier state_hash in
```

Comment:
I don't know if the `scan_state` and `pending_coinbases` would still be valid after the root changed. If they are still valid, we still want still want to find it using this function; if they are no longer valid, then this actually fixed a bug in bootstrap controller.